server/client.go

```diff
@@ -831,6 +832,22 @@ func (c *client) sendPing() {
 	c.sendProto([]byte("PING\r\n"), true)
 }
 
+// Generates the INFO to be sent to the client with the client ID included.
+// info arg will be copied since passed by value.
+// Assume lock is held.
+func (c *client) generateClientInfoJSON(info Info) []byte {
+	info.CID = c.cid
+	// Generate the info json
+	b, err := json.Marshal(info)
+	if err != nil {
+		c.Errorf("Error marshaling INFO JSON: %+v\n", err)
+		return nil
+	}
+	pcs := [][]byte{[]byte("INFO"), b, []byte(CR_LF)}
+	json := bytes.Join(pcs, []byte(" "))
```

Comment:
so that's instead of the fmt.Sprintf(), should we start doing that everywhere in new code? Is it really faster or less mem allocation?

Reply:
Definitely faster. Probably less allocs.